internal/wire/wire.go

```diff
@@ -643,12 +643,7 @@ func injectPass(name string, sig *types.Signature, calls []call, ig *injectorGen
 		}
 	}
 	if len(calls) == 0 {
-		for i := 0; i < params.Len(); i++ {
-			if types.Identical(injectSig.out, params.At(i).Type()) {
-				ig.p("\treturn %s", ig.paramNames[i])
-				break
-			}
-		}
+		ig.p("\treturn %s", ig.paramNames[set.For(injectSig.out).Arg().Index])
```

Comment:
Could `Arg` panic here due to some user error? If so, it might be worth to add a better error message.

Reply:
Any error here should be due to a bug: the only reason that the zero-call case should be able to occur is when the injector function result should be coming directly from an argument.